uberfire-workbench/uberfire-workbench-client-views-patternfly/src/main/java/org/uberfire/client/views/pfly/multipage/MultiPageEditorViewImpl.java

```diff
@@ -107,6 +107,10 @@ public int selectedPage() {
     @Override
     public void disablePage(int index) {
 
+        if (!isValid(index)) {
```

Comment:
This seems like an example of defensive programming. 

Is it fixing a symptom rather than a cause? Surely ```disablePage(..)``` and ```enablePage(..)``` should not be called if the ```index``` is greater than the number of pages setup?

Reply:
The `KieMultipleDocumentEditorWrapperView` is reinitialised depending on some events, and sometimes the `enableColumnsPage` is called with a fixed index (for the "Column tab") but the view has no tabs yet. Thus, I don't think that we can prevent this specific scenario where `IndexOutOfBoundsException` is raised.